File: TaskDistributionTest/TDT.Client/DistributedTaskState.cs
using Grpc.Core;
using MathCalc.OneArg;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace TDT.Client
{
    public class DistributedTaskState
    {
        public string CalcName { get; }
        public int WorkerCount { get; }
        public int BatchSize { get; }
        public Func<int, double> ArgFactory { get; }
        public EventHandler<MathCalcReply> ResponseCallback { get; }
        public AutoResetEvent RequestWaitHandle { get; }
        public AutoResetEvent ResponseWaitHandle { get; }
        public AsyncUnaryCall<MathCalcReply>[] CurrentTaskBatch { get; }
        public TaskCompletionSource<object> CompletionSource { get; } = new TaskCompletionSource<object>();

        public DistributedTaskState(string CalcName, int WorkerCount, int BatchSize, Func<int, double> ArgFactory, EventHandler<MathCalcReply> ResponseCallback)
        {
            this.CalcName = CalcName;
            this.WorkerCount = WorkerCount;
            this.BatchSize = BatchSize;
            this.ArgFactory = ArgFactory;
            this.ResponseCallback = ResponseCallback;
            CurrentTaskBatch = new AsyncUnaryCall<MathCalcReply>[WorkerCount];
            RequestWaitHandle = new AutoResetEvent(true);
            ResponseWaitHandle = new AutoResetEvent(false);

        }

    }
}

File: Projects/TaskDistributionTestJava/TDT.SpiceEngine/src/main/java/org/Main/DistributedTaskState.java
package org.Main;

import io.grpc.stub.StreamObserver;
import org.Main.Spice.MathCalc.OneArg.*;

import java.util.ArrayList;
import java.util.function.Function;

public class DistributedTaskState implements Runnable {
    public String calcName;
    public int workerCount;
    public int batchSize;
    public Function<Integer, Double> argFactory;
    public Function2<Object, MathCalcReply, Void> responseCallback;
    public AutoResetEvent requestWaitHandle;
    public AutoResetEvent responseWaitHandle;
    public ArrayList<StreamObserver<MathCalcReply>> currentTaskBatch;
//    public TaskCompletionSource<Object> completionSource = new TaskCompletionSource<Object>();

    public DistributedTaskState(String CalcName, int WorkerCount, int BatchSize, Function<Integer, Double> ArgFactory, Function2<Object, MathCalcReply, Void> ResponseCallback) {
        this.calcName = CalcName;
        this.workerCount = WorkerCount;
        this.batchSize = BatchSize;
        this.argFactory = ArgFactory;
        this.responseCallback = ResponseCallback;
        this.currentTaskBatch = new ArrayList<StreamObserver<MathCalcReply>>();
        requestWaitHandle = new AutoResetEvent(true);
        responseWaitHandle = new AutoResetEvent(false);
    }

    public void run() {

    }
}
